src/Microsoft.TestPlatform.ObjectModel/Client/Interfaces/IWarningLogger.cs

```diff
@@ -0,0 +1,19 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+
+#nullable disable
+
+namespace Microsoft.VisualStudio.TestPlatform.ObjectModel.Client;
+
+/// <summary>
+/// DO NOT use this to extend TestPlatform, it is public only because ITestPlatform is also public, and will be made internal later.
+/// </summary>
+public interface IWarningLogger
```

Comment:
Why not making a "general" logger? If we now need to flow an Info (or maybe an Error) we will have to mimic this behavior instead of simply adding a new API to this existing type.

Reply:
I was thinking to make this just some *Logger, but I feel like we have that concept somewhere already, and this interface should hopefully not grow. I would have used IBaseTestEventRegistrar or how that interface is called, but it is not defined in object model, so this just re-implements it.

It would be much better to give a more general IUserMessageLogger to the instances where we need it instead of flowing a logger here via method parameters. That is a concept that should live in ObjectModel and not ITestPlatform.